lib/grape-swagger/doc_methods/data_type.rb

```diff
@@ -55,7 +55,16 @@ def parse_entity_name(model)
           elsif model.respond_to?(:name)
             model.name.demodulize.camelize
           else
-            model.to_s.split('::').last
+            klass = begin
```

Comment:
This seems unnecessarily complicated. Why doesn't `model` just implement `.data_type` for example?

Reply:
when using a custom types like the SecureUri example here: https://github.com/ruby-grape/grape#custom-types-and-coercions
Model comes into this function as a string and not an object. This is probably me missing something critical if this seems silly.

Reply:
In line 37 above, in `parse_multi_type(raw_data_type)` they basically do the same thing                       `Object.const_get(model, false)` and `rescue NameError`